scalableminds/webknossos
# Uglify

Test to include `uglify-js` in our production build pipeline to shrink our bundles file size even more.
I've also changed the source maps options, which may take a little longer to generate now, but are much! nicer. You should be able to see the original import statements and set in-line debugging breakpoints :)
The webpack build time for production went up by ~60s I think, but we gain a pretty hefty reduction in file size from that (vendor.js is down from 5Mb to 2Mb, gzipped from 1Mb to 500Kb), not sure about the trade-off there.
~Webpack build time in development should not really be affected, apart from the different source maps option.~
Initial webpack build time in development goes up from ~17s to ~21s using the different source maps option.

### Steps to test:
- Check out the branch, change some code and reload the page to see whether webpack is still fast enough.

### Issues:
- makes webknossos load faster and debugging easier
- fixes #2218

------
